conjure/servicewriter.go

```diff
@@ -1018,26 +1052,17 @@ func returnTypesForEndpoint(endpointDefinition spec.EndpointDefinition, info typ
 	var returnTypes []expression.Type
 	imports := make(StringSet)
 	if endpointDefinition.Returns != nil {
-		var goType string
-		returnBinary, err := isReturnTypeSpecificType(endpointDefinition.Returns, visitors.IsBinary)
+		typer, err := visitors.NewConjureTypeProviderTyper(*endpointDefinition.Returns, info)
 		if err != nil {
 			return nil, nil, err
 		}
-		if returnBinary {
-			// special case: "binary" type resolves to []byte in structs, but indicates a streaming response when
-			// specified as the return type of a service, so use "io.ReadCloser".
-			goType = types.IOReadCloserType.GoType(info)
-			imports.AddAll(NewStringSet(types.IOReadCloserType.ImportPaths()...))
-		} else {
-			typer, err := visitors.NewConjureTypeProviderTyper(*endpointDefinition.Returns, info)
-			if err != nil {
-				return nil, nil, err
-			}
-			goType = typer.GoType(info)
-			if err != nil {
-				return nil, nil, errors.Wrapf(err, "failed to process return type %q", goType)
-			}
-		}
+
+		// special case: "binary" type resolves to []byte in structs, but indicates a streaming response when
+		// specified as the return type of a service, so replace all nested references with "io.ReadCloser".
```

Comment:
We spoke about this on call and Slack, but forgot the final conclusion: are all nested references replaced (including in lists/sets)? If so, is that correct? If not, comment should be updated because it doesn't describe reality.

Reply:
Good catch, I revisited the old code where I generated that `[][]binary` and it now generated `[]ReadCloser` which seems wrong. What I've done is updated this map function to only map `binary` and `optional<binary>` which means that `iterable<binary>` types will be specified as `[][]byte` in the api. I don't think this is correct but it isn't a regression so I'll file an issue to fix it at a later date. I did an internal search and found zero usages of `list<binary>` and only test usages of `set<binary>` which helps explain how this hasn't been an issue thus far.

Reply:
Here's that issue: https://github.com/palantir/conjure-go/issues/194